rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -252,25 +271,55 @@ def get_state(self) -> Mapping[str, Any]:
 
     def add_module(
         self,
+        *,
         module_id: ModuleID,
         module_cls: Type[RLModule],
         module_kwargs: Mapping[str, Any],
+        set_optimizer_fn: Optional[Callable[[RLModule], ParamOptimizerPairs]] = None,
     ) -> None:
         """Add a module to the trainer.
 
         Args:
             module_id: The id of the module to add.
             module_cls: The module class to add.
             module_kwargs: The config for the module.
+            set_optimizer_fn: A function that takes in the module and returns a list of
```

Comment:
It's a complex behavior to be finding the first added optimizer in the optimizer_to_param dictionary. We should either have the user explicitly define a default for us, or pass one themselves.

This is going to create a lot of cognitive load for the user in figuring out which optimizer class was used to create the agent. Of course unless the optimizer that was used is the same for all the modules, which is very likely.

Reply:
Any automated inference here would put a cognitive load on the user, and we need to clarify that in the docstring. I think being explicit is good here as well. Just asking the user to say what optimizer they want to use for the new added module. That's also easy to specify. I am gonna go with being fully explicit here.

Reply:
ok fixed this by introducing an extra optional parameter called optimizer_cls. Users have to either provide optimizer_cls for the new parameters or define a function that returns both optimizers and their corresponding parameters with more flexibility.